ui/app/components/job-status/panel.js

```diff
@@ -1,7 +1,105 @@
 // @ts-check
 import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { task } from 'ember-concurrency';
+import messageFromAdapterError from 'nomad-ui/utils/message-from-adapter-error';
+import { alias } from '@ember/object/computed';
 
 export default class JobStatusPanelComponent extends Component {
+  @alias('args.job') job;
+  @alias('args.handleError') handleError = () => {};
+
+  // #region deployment
+  @tracked oldVersionAllocBlockIDs = [];
+
+  // Called via did-insert; sets a static array of "outgoing"
+  // allocations we can track throughout a deployment
+  establishOldAllocBlockIDs() {
+    this.oldVersionAllocBlockIDs = this.job.allocations.filter(
+      (a) =>
+        a.clientStatus === 'running' &&
+        a.jobVersion !== this.deployment.get('versionNumber')
+    );
```

Comment:
Maybe I'm misunderstanding the question, but `establishOldAllocBlockIDs` and `oldVersionAllocBlockIDs` (and any downstream dependent properties) are only used when `isActivelyDeploying` is true. This panel has another mode (not actively deploying), where these properties are not used.

While this speaks to maybe splitting this into multiple components instead of a single panel, I'm not sure where unmount cleanup would come into play